src/main/ngapp/accio/src/app/editor/editor.component.ts

```diff
@@ -302,32 +300,70 @@ export class EditorComponent implements OnInit {
       this.maskImageData.data[i + 3] = 255 - this.maskImageData.data[i+ 3];
     }
     this.drawMask();
+    if (this.maskTool == MaskTool.magicWand) {
+      this.disableFloodFill = false;
+    }
     this.disableSubmit = false;
-    if (this.maskTool == 'magic-wand') {
-      this.disableFloodFill = false
+  }
+
+
+  /**  Retrieves new tolerance value from child component toolbar and updates. */
+  updateTolerance(value: number) {
+    this.tolerance = value;
+    console.log('new tolerance: ' + value);
+  }
+
+
+ /** 
+  *  Retrieves new alpha value from child component toolbar and draws mask with new alpha.
+  *  The alpha value cannot be larger than 1 or less than 0, so the value is adjusted to fit in range.
+  */
+  updateMaskAlpha(value: number) {
+    this.maskAlpha = (Math.abs(value) % 1);
```

Comment:
I don't think this is exactly what you want here as 1.2 % 1 will be 0.2  I think instead you want to clamp so:

min(max(value, 0.0), 1.0)

Reply:
Did you forget to push these new changes?  Or am I thinking about this wrong?